src/Simplex/Chat.hs

```diff
@@ -1655,6 +1655,11 @@ processChatCommand' vr = \case
     case conn'_ of
       Just conn' -> pure $ CRConnectionIncognitoUpdated user conn'
       Nothing -> throwChatError CEConnectionIncognitoChangeProhibited
+  APISetConnectionUserId connId oldUserId -> withUser $ \user@User {userId} -> do
+    conn' <- withFastStore $ \db -> do
+      conn <- getPendingContactConnection db oldUserId connId
+      liftIO $ updatePendingContactConnectionUserId db userId oldUserId conn
+    pure $ CRConnectionUserIdUpdated user conn'
```

Comment:
I think we should be calling this api while being in the current user the connection was initially created for:

```suggestion
  APISetConnectionUserId connId newUserId -> withUser $ \user@User {userId} -> do
    conn' <- withFastStore $ \db -> do
      conn <- getPendingContactConnection db userId connId
      liftIO $ updatePendingContactConnectionUserId db userId newUserId conn
    pure $ CRConnectionUserIdUpdated user conn'
```

The main reason being that we shouldn't switch current user in ui before calling this api for these reasons:
- when we do switch after api returns successful response, connection will already be in that user's chat list via apiGetChats, without any additional movements needed
- this api may fail with error (as all db queries can theoretically fail as it's IO), and connection would stay in current user, in which case UI shouldn't even switch to new user - and if it switches before api completes we're left with strange state there with connection being in old user, while view trying to show it for new user (will likely have artifacts of its own)

Reply:
we also have to create a new agent api to set new user id and call it here, as by this point connection record is already created in agent with old userId (see APIAddContact, withAgent createConnection). note that agent uses its own ids (for historical reasons), so new connection user has to be read from db and userId retrieved like so from it: `aUserId newUser`

Reply:
actually we also have to consider re-creating on that user's configured servers.. let's discuss